src/components/button-swap/button-swap.stories.mdx

```diff
@@ -0,0 +1,163 @@
+import { Story, Canvas, Meta } from '@storybook/addon-docs/blocks';
+import { makeTwigInclude } from '../../make-twig-include';
+import { useEffect } from '@storybook/client-api';
+import buttonSwap from './button-swap.twig';
+import { initButtonSwap } from './button-swap.ts';
+// Helper function to initialize toggling button JS
+const buttonSwapStory = (args) => {
+  useEffect(() => {
+    const buttonSwapEls = [
+      ...document.querySelectorAll('.js-c-button-swap'),
+    ].map((buttonSwapEl) =>
+      initButtonSwap(buttonSwapEl, {
+        firstBtnCallback: () => console.log('Subscribed!'),
+        secondBtnCallback: () => console.log('Unsubscribed!'),
+      })
+    );
+    return () =>
+      buttonSwapEls.forEach((buttonSwapEl) => buttonSwapEl.destroy());
+  });
+  return buttonSwap(args);
+};
+
+<Meta
+  title="Components/Button Swap"
+  argTypes={{
+    content_start_icon: {
+      type: { name: 'string' },
+      control: {
+        type: 'select',
+        options: ['bell', 'magnifying-glass'],
+      },
+    },
+    first_button_visually_hidden_text: { type: { name: 'string' } },
+    first_button_label: { type: { name: 'string' } },
+    second_button_visually_hidden_text: { type: { name: 'string' } },
+    second_button_label: { type: { name: 'string' } },
+  }}
+/>
+
+# Button Swap
+
+The Button Swap component was built with the article Push Notifications use case in mind. It has two buttons that are swapped on click representing two visual states accompanied with visually hidden text for a more inclusive experience.
+
+You have the ability to pass `firstBtnCallback` and `secondBtnCallback` callback functions to the `initButtonSwap` initialization function to add custom actions for when each of the buttons is clicked.
+
+<Canvas>
+  <Story
+    name="Default"
+    inline={false}
+    args={{
+      first_button_visually_hidden_text: 'Notifications have been turned off.',
+      first_button_label: 'Give me all the notifications!',
+      second_button_visually_hidden_text: 'Notifications have been turned on.',
+      second_button_label: 'Just kidding, stop notifications, please.',
```

Comment:
These arguments are kind of a mouthful! I wonder if they could be simplified? Idea:

```
initial_label
initial_visual_label
swapped_label
swapped_visual_label
```

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1516/commits/3da46ecee1dbf6b294f0c9113681ee7ef8f4942b